internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1753,6 +1807,8 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 
 			existing.Model.Properties.NetworkProfile.NatGatewayProfile = &natGatewayProfile
 		}
+
+		existing.Model.Properties.NetworkProfile.KubeProxyConfig = expandKubernetesClusterKubeProxyConfig(d.Get("network_profile.0.kube_proxy").([]interface{}))
```

Comment:
This needs to be wrapped in `d.HasChanges()` so that the default for existing clusters is respected.